eras/shelley-ma/test-suite/test/Test/Cardano/Ledger/Mary/Value.hs

```diff
@@ -354,6 +361,85 @@ compactRoundTrip = forAll gen $ \v ->
 compactTest :: TestTree
 compactTest = testProperty "fromCompact . toCompact == id" compactRoundTrip
 
+coinCBORTests :: TestTree
+coinCBORTests =
+  testGroup
+    "Coin CBOR roundtrips"
+    [ testCase "decode 0 as Coin for pre-conway" $
+        forM_ [natVersion @2 .. natVersion @8] $ \version ->
```

Comment:
There is no nee to supply versions here, `roundTripVersionExpectation` already goes through all versions
```suggestion
        roundTripVersionExpectation minBound (natVersion @8) (Coin 0)
```